flink-staging/flink-streaming/flink-streaming-scala/src/test/java/org/apache/flink/streaming/scala/api/CsvOutputFormatITCase.java

```diff
@@ -36,6 +37,10 @@ protected void preSubmit() throws Exception {
 	@Override
 	protected void testProgram() throws Exception {
 		OutputFormatTestPrograms.wordCountToCsv(WordCountData.TEXT, resultPath);
+		OutputFormatTestPrograms.wordCountToCsv(WordCountData.TEXT, resultPath, 1);
```

Comment:
It is expected that the test fails if you set the overwrite mode to `NO_OVERWRITE` because you start multiple programs that all write to the same location.

I debugged the issue and found that the `ForkableFlinkMiniCluster` sets the overwrite mode to OVERWRITE. That explains the behavior. However, this also means that you cannot test the correctness of your methods by setting the overwrite mode to OVERWRITE because this is the default behavior.

I would change the `CsvOuputFormatITCase` to extend `StreamingMultipleProgramsTestBase` and run each program in a dedicated test method.